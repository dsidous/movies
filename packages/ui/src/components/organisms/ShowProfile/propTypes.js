import { array, object, func } from 'prop-types';

export const propTypes = {
  config: object.isRequired,
  show: object.isRequired,
  dcolor: array.isRequired,
  handleShowClick: func.isRequired,
  handleFullCrewClick: func.isRequired,
};

export const defaultProps = {};
