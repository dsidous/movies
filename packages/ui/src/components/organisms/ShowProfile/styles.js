import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    backdrops: {
      display: 'none',

      [theme.breakpoints.up('sm')]: {
        display: 'block',
      },
    },

    header: {
      borderBottom: theme.separator,
      color: '#fff',
      display: 'block',
      position: 'relative',

      [theme.breakpoints.up('sm')]: {
        borderTop: theme.separator,
      },

      '&:before': {
        backgroundImage: ({ backdropURL }) => `url(${backdropURL})`,
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat',
        backgroundPosition: '50% 50%',
        content: '" "',
        display: 'block',
        filter: 'brightness(120%) grayscale(1)',
        height: '100%',
        left: 0,
        position: 'absolute',
        right: 0,
        width: '100%',
        zIndex: '-1',
      },

      '&:after': {
        backgroundImage: ({ dcolor }) =>
          `radial-gradient(at 10% 30%,
          rgb(${dcolor[0]},${dcolor[1]},${dcolor[2]}) 0%, 
          #342931 100%
        )`,
        content: '" "',
        display: 'block',
        filter: 'opacity(92%)',
        height: '100%',
        left: 0,
        position: 'absolute',
        right: 0,
        top: 0,
        transition: 'filter 1s',
        zIndex: '-1',
        width: '100%',
        willChange: 'opacity',
      },
    },

    headerInner: {
      display: 'grid',
      gridGap: 15,
      gridTemplateColumns: '1fr 2fr',
      gridTemplateAreas:
        '"mdata mdata" "mposter moverview" ". mtrailer" "mcrew mcrew"',
      padding: '20px 0 40px',
      position: 'relative',

      [theme.breakpoints.up('sm')]: {
        gridColumnGap: 40,
        gridTemplateColumns: 'minmax(200px, 1fr) minmax(400px, 2fr)',
        gridTemplateRows: 'auto',
        gridTemplateAreas:
          '"mposter mdata" "mposter moverview" "mposter mcrew" "mposter mtrailer" "mcast mcast"',
        margin: '0 auto',
        maxWidth: theme.breakpoints.values.lg,
        padding: theme.spacing(5, 2),
      },
    },

    tagline: {
      display: 'none',
    },

    poster: {
      gridArea: 'mposter',
      gridRow: '2 / 4',
      marginLeft: 10,
      position: 'relative',

      [theme.breakpoints.up('sm')]: {
        gridRow: '1/5',
      },

      '& img': {
        borderRadius: 2,
        boxShadow: '0 0 20px -5px #000',
        width: '100%',
      },
    },

    noImage: {
      background: '#666',
      color: '#444',
      font: '80px/513px FontAwesome',
      height: 513,
      textAlign: 'center',
      width: 342,

      '&:before': {
        content: '"\f03e"',
      },

      '&.smaller': {
        font: '60px/231px FontAwesome',
        height: 231,
        width: 154,
      },
    },

    data: {
      gridArea: 'mdata',
      fontSize: 12,
      padding: 10,

      [theme.breakpoints.up('sm')]: {
        fontSize: '1.2em',
        padding: 0,
      },
    },

    title: {
      color: '#fff',
      fontSize: 20,
      fontWeight: theme.fontWeightLight,
      letterSpacing: 1,
      margin: 0,

      [theme.breakpoints.up('sm')]: {
        fontSize: '2.3em',
        margin: '20px 0 0',
      },
    },

    rating: {
      border: 'none',
      color: '#fff',
      fontSize: 22,
      marginLeft: 20,
      paddingLeft: 23,
      position: 'relative',

      '&::before': {
        font: '20px/20px FontAwesome',
        color: '#e3cf7a',
        content: '"\\f005"',
        left: 0,
        position: 'absolute',
        top: 2,
      },

      '&::after': {
        fontSize: 12,
        bottom: 2,
        color: '#bbb',
        content: '"/10"',
        right: 'auto',
        top: 'auto',
      },
    },

    titleTags: {
      listStyle: 'none',
      margin: 0,
      padding: 0,

      '& li': {
        display: 'inline-block',
        padding: '0 10px',
        position: 'relative',

        '&:first-child': {
          paddingLeft: 0,
        },

        '&:not(:last-child):after': {
          content: '"\\2022"',
          position: 'absolute',
          right: '-4px',
          top: 0,
        },
      },
    },

    genres: {
      marginLeft: 0,
      padding: 0,

      '& li': {
        border: 'none',
        padding: '0 2px',

        '&:not(:last-child):after': {
          content: '","',
          position: 'relative',
          right: 0,
        },

        '&:last-child:after': {
          content: '" "',
        },
      },
    },

    overview: {
      gridArea: 'moverview',
      marginRight: 10,

      '& h4': {
        marginTop: 0,
      },
    },
  }),
  { name: 'show' },
);
