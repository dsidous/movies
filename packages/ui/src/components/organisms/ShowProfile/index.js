/* eslint-disable camelcase */
import React from 'react';
import Style from 'style-it';

import { propTypes } from './propTypes';
import PlayTrailer from '../../atoms/PlayTrailer';
import TopCast from '../../molecules/TopCast';
import Crew from '../../atoms/Crew';
import SimilarMovies from '../SimilarMovies';
import FullScreenBackdrop from '../../atoms/FullScreenBackdrop';
import Reviews from '../../molecules/Reviews';
import SEO from '../../atoms/SEO';
import WatchlistBookmark from '../../atoms/WatchlistBookmark';
import LastSeason from '../../molecules/LastSeason';

const ShowProfile = ({
  config: {
    images: { secure_base_url, poster_sizes, backdrop_sizes },
  },
  show: {
    id,
    backdrop_path,
    poster_path,
    genres,
    name,
    title,
    release_date,
    first_air_date,
    reviews,
    runtime,
    vote_average,
    tagline,
    overview,
    images: { backdrops },
    videos,
    similar,
    credits: { cast, crew },
    seasons,
  },
  dcolor,
  show,
  handleFullCrewClick,
  handleShowClick,
}) => {
  const showDate = first_air_date || release_date;
  const showTitle = title || name;
  let lastSeason = null;

  if (seasons) {
    [lastSeason] = seasons.sort((a, b) => b.season_number - a.season_number);
  }

  const posterURL = secure_base_url + poster_sizes[3] + poster_path;
  const backdropURL = secure_base_url + backdrop_sizes[1] + backdrop_path;
  const video = videos
    ? videos.filter(element => element.type === 'Trailer')[0]
    : [];

  const genres_html = genres.map(genre => (
    <li key={genre.id} className="movie-genres">
      {genre.name}
    </li>
  ));

  return (
    <div>
      <Style>
        {`
        .main-header:after {
          background-image: radial-gradient(at 10% 30%, rgb(${dcolor[0]},${dcolor[1]},${dcolor[2]}) 0%, #342931 100%);
        }
        .main-header:before {
          background-image: url(${backdropURL});
        }
      `}
      </Style>
      <SEO title={showTitle} />
      <div className="full-background">
        {backdrops[0] && (
          <FullScreenBackdrop
            backdrops={backdrops.map(
              image =>
                `${secure_base_url}${backdrop_sizes[2]}${image.file_path}`,
            )}
          />
        )}
      </div>

      <div className="main-header">
        <div className="main-header-inner">
          <div className="poster">
            {poster_path !== null ? (
              <img src={posterURL} alt="poster" />
            ) : (
              <div className="movie-no-image-holder" />
            )}
            <WatchlistBookmark movie={show} />
          </div>
          <div className="movie-data">
            <h1 className="movie-title">
              {showTitle}
              <span className="movie-rating">{vote_average}</span>
            </h1>

            <ul className="title-tags">
              <li>{showDate.slice(0, 4)}</li>
              {runtime && <li>{`${runtime} min`}</li>}
              <li>
                <ul className="title-tags__genres">{genres_html}</ul>
              </li>
            </ul>

            <h4 className="movie-tagline">{tagline}</h4>
          </div>

          <div className="overview">
            <h4>Overview</h4>
            {overview}
          </div>

          {video && <PlayTrailer video={video} />}

          {crew[0] && <Crew crew={crew.slice(0, 4)} />}
        </div>
      </div>
      {cast[0] && (
        <TopCast
          cast={cast.slice(0, 6)}
          handleFullCrewClick={handleFullCrewClick}
        />
      )}
      {lastSeason && <LastSeason tvId={id} season={lastSeason} />}

      {reviews && reviews[0] && <Reviews reviews={reviews} />}

      {similar.results[0] && (
        <SimilarMovies
          similar={similar.results}
          handleMovieClick={handleShowClick}
        />
      )}
    </div>
  );
};

ShowProfile.propTypes = propTypes;

export default ShowProfile;
