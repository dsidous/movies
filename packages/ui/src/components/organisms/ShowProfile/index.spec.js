import React from 'react';
import { shallow } from 'enzyme';
import ShowProfile from '.';

const mockProps = {
  config: {
    images: {
      secure_base_url: 'path/to/image',
      poster_sizes: ['1', '2', '3', '4'],
      backdrop_sizes: ['1', '2', '3', '4'],
    },
  },
  show: {
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    genres: [
      {
        id: 1,
        genre_name: 'genre',
      },
    ],
    name: 'name',
    release_date: '01.01.2010',
    reviews: {},
    runtime: 'runtime',
    vote_average: 1,
    tagline: 'tagline',
    overview: 'overview',
    images: {
      backdrops: [{ file_path: 'path/to/image' }],
    },
    videos: [{ type: 'trailer' }],
    similar: {
      results: [],
    },
    credits: {
      cast: [],
      crew: [],
    },
    seasons: [],
  },
  dcolor: [1, 2, 3],
  handleFullCrewClick: jest.fn(),
  handleShowClick: jest.fn(),
};

describe('Organisms/ShowProfile', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<ShowProfile {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
