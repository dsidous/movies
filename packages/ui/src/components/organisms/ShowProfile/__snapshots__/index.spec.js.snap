// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/ShowProfile should render as expected 1`] = `ShallowWrapper {}`;
